Source/AdUnits/Native/NativeAd.swift

```diff
@@ -0,0 +1,320 @@
+                return nil
+            }
+            let ad:NativeAd = NativeAd()
+            //assets
+            if let assets = json["assets"] as? [AnyObject] {
+                for adObject in assets {
+                    if let adObject = adObject as? [String : AnyObject]{
+                        //title
+                        if let title = adObject["title"], let text = title["text"] as? String {
+                            ad.title = text;
+                        }
+                        //description
+                        if let id = adObject["id"] as? Int, id == 3, let description = adObject["data"], let text = description["value"] as? String {
+                            ad.text = text;
+                        }
+                        //sponsoredBy
+                        if let id = adObject["id"] as? Int, id == 5, let sponsoredBy = adObject["data"], let text = sponsoredBy["value"] as? String {
+                            ad.sponsoredBy = text;
+                        }
+                        //call to action
+                        if let id = adObject["id"] as? Int, id == 4, let callToAction = adObject["data"], let text = callToAction["value"] as? String {
+                            ad.callToAction = text;
+                        }
+                        if let img = adObject["img"]{
+                            if let id = adObject["id"] as? Int, id == 2{
+                                //img
+                                if let url = img["url"] as? String {
+                                    ad.imageUrl = url;
+                                }
+                            }else{
+                                //icon
+                                if let url = img["url"] as? String {
+                                    ad.iconUrl = url;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            
+            let price = response["price"] as? Double
+            
+            //clickUrl
+            if let link = json["link"] as? [String : AnyObject], var url = link["url"] as? String {
+                if (url.contains("{AUCTION_PRICE}") && price != nil){
+                    url = url.replacingOccurrences(of: "{AUCTION_PRICE}", with: "\(price ?? 0.0)")
+                }
+                ad.clickUrl = url;
+            }
+            //eventtrackers
+            if let eventtrackers = json["eventtrackers"] as? [AnyObject] {
+                for eventtracker in eventtrackers {
+                    if let eventtracker = eventtracker as? [String : AnyObject], var url = eventtracker["url"] as? String  {
+                        if (url.contains("{AUCTION_PRICE}") && price != nil){
+                            url = url.replacingOccurrences(of: "{AUCTION_PRICE}", with: "\(price ?? 0.0)")
+                        }
+                        ad.impTrackers.append(url)
```

Comment:
Two questions here. 

1. The **type** of event tracker could be different from the Impression, but all of them will be tracked at the same time according to the current logic. Does it make sense to add a filter for the tracker type before adding the url to the EventType?

2. The **method** of the event tracker could be javascript. It means that the url will just contain the js file that should be _executed_ eventually, but not only requested, isn't it?  Maybe prebid SDK should not support js event trackers?

Reply:
Hi, please find the answers below: 
1. for event trackers, current native ad request only request for impression trackers, server should not return a different type of tracker
2. for in-app native, we should not support js event trackers, if needed we can send signals to the prebid server

Reply:
Thanks for the clarification!

In this case enums EventType and EventTracking look confusing. Looking at them, the publisher can assume that he is able to request such event trackers and think that they will be processed with prebid. It would be helpful to add some comments around.   

I can't find the filtering logic on the server respectively to the mrc event types.
Here we can see only validation for the range of possible values but all iab events should be passed.
https://github.com/prebid/prebid-server/blob/9c2e6301f098262cf1cade92b1270c5546f40918/endpoints/openrtb2/auction.go#L728 
Also, this logic, unfortunately, looks incorrect as well. Prebid server should support events greater than 500 to request custom event trackers. For example, the Open Measurement framework requires the event tracker with type 555.

Reply:
Hi @YuriyVelichkoPI

I looked at the code again, the event type and tracking code was written before for in banner native. We're utilizing it for in app native. I'll add it in the docs that only image impression trackers are supported for now.

As for prebid server issues, if you think it's incorrect, would you mind opening up an issue in that repo?

Reply:
@anwzhang  Got it. Thank you! 

Sure, we at OpenX, are going to make additional research around supported types and open the issue for PBS with particular motivation and details.